src/kernels/jupyter/jupyterUriProviderRegistration.ts

```diff
@@ -53,14 +61,35 @@ export class JupyterUriProviderRegistration implements IJupyterUriProviderRegist
     }
 
     private async loadOtherExtensions(): Promise<void> {
+        const extensionIds = new Set<string>();
+        this.globalMemento
```

Comment:
In the case of `github extensions`, they can register the lazy proxy controller when a notebook opens and they register the provider with us.
When we reload vscode, we try to get the provider information, however its possible github or other extensions haven't activated yet, for such extensions we don'e need an entry in package.json for `pythonRemoteServerProvider`, instead we can just keep track of the extensions that own a specific provider and load and activate those extensions.